### PR TITLE
fix(payments): update ui and verbiage for payment bubbles

### LINKED ISSUE #327: feat(Pay): Improve UI for payment request (chat message)
We should improve the appearance of the chat message containing the payment request. 

Currently it looks like this:
Screenshot 2024-07-31 at 8 19 41 AM

### LINKED ISSUE #663: edit(UI): Payments UI Improvement
Currently, there is no way to request funds in the payments UI, and the popup is a bit crowded. 
### **New payment UI:**
Send/ Request popup

Send UI

Request UI

Request Message

### PR DESCRIPTION
**Pull request by Jekrimo**
### What this PR does 📖

-update ui and verbiage for payment bubbles

### Which issue(s) this PR fixes 🔨

- Resolve #327

### CODE REVIEW

**Review by InfamousVague — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
@phillsatellite and i were testing and we have the following feedback

browser: chrome


| Issue Description | Suggested Fix | Screenshot/Video |
|-------------------|---------------|------------------|
| When a user sends a request, it says "Send request," but it should say "Request has been sent." The current wording also appears like a clickable button. | Update the text to "Request has been sent" and ensure it does not resemble a clickable button. | Screenshot 1 |
| The request notification is missing information about the coin and amount. | Include details of the coin and amount in the request notification. First screenshot is from the PR, second screenshot is from dev. | Screenshot 2 <br> Screenshot 2 - dev |
| When User A sends a request to User B, and User B cancels it, `/reject` appears in the sidebar preview. | Replace `/reject` in the sidebar with a clearer status message, such as "Request Rejected." | Screenshot 3 |
| When User A sends a request to User B, and User A cancels it, "Payment canceled" appears, which is misleading. There’s also an additional message saying "You canceled." | Change "Payment canceled" to a clearer status like "Request canceled by you." Consider consolidating this message to update the original notification instead of adding a new message. | Screenshot 4 |
| User A sends a request to User B, and User B cancels it, resulting in an empty message for User A. | Ensure a descriptive message is shown to User A when User B cancels the request, e.g., "User B canceled your request." | Screenshot 5 |
| Sometimes clicking on send coin returns error on console but no error on the UI. | Display a user-friendly error message on the UI when a console error occurs, or resolve the underlying error if possible. | Screenshot 6 |
| If you get a request, click it and then exit without sending, and then try to click send coin again, it will not do anything. | Ensure that exiting without sending does not "lock" the Send Coin button. Reset the button state when the request window is closed without completion. | N/A |
| Sometimes a user is not able to send any requests for ETH or BTC; the request button just doesn't change to clickable. | Investigate and fix the issue preventing the request button from becoming clickable, ensuring it updates appropriately based on user action. | N/A |
| When clicking on pay for BTC, it opened the Xverse app, but for ETH, it just gives an error in the console. | Make the pay functionality consistent across currencies, so that clicking pay for ETH initiates the appropriate action without console errors. | Screenshot |

File: src/lib/lang/en.json [hunk 1]
```diff
@@ -202,8 +202,15 @@
         "to": "To",
         "note": "Note",
         "date": "Date",
+        "decline": "Decline",
         "create": "Create Payment Request",
+        "decline_payment": "Decline Payment",
```
Comment (lgmarchi) [inline src/lib/lang/en.json]: Jeff, we don't use snake_case on Uplink, we use cammelCase. Could you change please?

File: src/lib/utils/Wallet.ts [hunk 2]
```diff
@@ -489,6 +491,11 @@ export class Transfer {
         return `/request ${transfer}`
     }
 
+    torejectString(id: string) {
```
Comment (lgmarchi) [inline src/lib/utils/Wallet.ts]: toRejectString would be better. With R.

File: src/lib/utils/Wallet.ts [hunk 3]
```diff
@@ -501,20 +508,32 @@ export class Transfer {
...
         let transfer = new Transfer()
         try {
             let parsed = JSON.parse(json, (k, v) => (k === "amount" && typeof v === "string" ? BigInt(v) : v))
             transfer.asset = parsed.asset
             transfer.amount = parsed.amount
             transfer.toAddress = parsed.toAddress
             transfer.amountPreview = parsed.amountPreview
-        } catch {
-            // if it wasn't valid json, do nothing.
+        } catch {}
```
Comment (lgmarchi) [inline src/lib/utils/Wallet.ts]: Let's add a log here, just to let us know that JSON.parse failed.

File: src/lib/utils/Wallet.ts [hunk 4]
```diff
@@ -501,20 +508,32 @@ export class Transfer {
...
+    } else if (msg.startsWith(rejectCmd)) {
+        let json = msg.substring(rejectCmd.length, msg.length)
+        let transfer = new Transfer()
+        try {
+            let parsed = JSON.parse(json, (k, v) => (k === "amount" && typeof v === "string" ? BigInt(v) : v))
+            transfer.asset = parsed.asset
+            transfer.amount = parsed.amount
+            transfer.toAddress = parsed.toAddress
+            transfer.amountPreview = parsed.amountPreview
+        } catch {}
```
Comment (lgmarchi) [inline src/lib/utils/Wallet.ts]: Same thing here about error log. Or at least warning log if it not so bad.

File: src/routes/chat/+page.svelte [hunk 5]
```diff
@@ -163,7 +150,7 @@
         })
     }
 
-    function build_context_items(message: Message, file?: Attachment) {
+    function build_context_items(message: MessageType, file?: Attachment) {
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: Let change to cammelCase my friend. buildContextItems.

File: src/routes/chat/+page.svelte [hunk 6]
```diff
@@ -285,6 +278,52 @@
         })
     }
 
+    $: rejectedPayments = Store.state.paymentTracker
+
+    async function sendPaymentMessage(message: MessageType, paymentType: string) {
+        let transfer = new Transfer()
+        let chat = get(Store.state.activeChat)
+        let rejectTranser = transfer.torejectString(message.id)
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: Transer is wrong, change to Transfer.

File: src/routes/chat/+page.svelte [hunk 7]
```diff
@@ -285,6 +278,52 @@
...
+                if (getValidPaymentRequest(message.text[0])) {
+                    getValidPaymentRequest(message.text[0])?.execute()
+                }
+                Store.state.paymentTracker.update(payments => {
+                    const alreadyRejected = payments.some(payment => payment.messageId === message.id)
+
+                    if (!alreadyRejected) {
+                        return [...payments, { messageId: message.id, senderId: message.details.origin, rejectedPayment: false }]
+                    } else {
+                        console.log(`MessageId ${message.id} is already in the rejected payments list`)
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte:321]: Let's use our Logger here my friend.

File: src/routes/chat/+page.svelte [hunk 8]
```diff
@@ -285,6 +278,52 @@
...
+        if (paymentType === "reject") {
+            let result = await RaygunStoreInstance.send(chat.id, txt, [])
+            result.onSuccess(res => {
+                Store.state.paymentTracker.update(payments => {
+                    const alreadyRejected = payments.some(payment => payment.messageId === message.id)
+
+                    if (!alreadyRejected) {
+                        return [...payments, { messageId: message.id, senderId: message.details.origin, rejectedPayment: true }]
+                    } else {
+                        console.log(`MessageId ${message.id} is already in the rejected payments list`)
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte:338]: Use our Logger here.

File: src/routes/chat/+page.svelte [hunk 9]
```diff
@@ -312,7 +351,28 @@
         }, 500)
     })
 
-    function getPinned(conversation: ConversationMessages | undefined): Message[] {
+    function checkForActiveRequest(message: MessageType, messageLine: string) {
+        const idMatch = messageLine.match(/^\/reject\s([a-f0-9-]{36})$/)
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte:376]: We can put this regex in a const variable in utils?

File: src/routes/chat/+page.svelte [hunk 10]
```diff
@@ -744,7 +804,47 @@
                                                 {#if editing_message === message.id}
                                                     <Input hook="chat-message-edit-input-{editing_message}" alt bind:value={editing_text} autoFocus rich on:enter={_ => edit_message(message.id, editing_text ? editing_text : "")} />
                                                 {:else}
-                                                    <MessageText chat={$activeChat.id} texts={message.text} remote={group.details.remote} type={message.type} />
+                                                    {#each message.text as line}
+                                                        {#if line.startsWith("/reject")}
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: My friend, we use a lot of "/reject" in our code, and "/request" as well. What do you think to put those values in const variables?

File: src/routes/chat/+page.svelte [hunk 11]
```diff
@@ -744,7 +804,47 @@
...
+                                                            {#if !checkForActiveRequest(message, line)}
+                                                                {#if $own_user.key !== message.details.origin}
+                                                                    <Text hook="text-chat-message" markdown={"Payment Rejected"} appearance={group.details.remote ? Appearance.Default : Appearance.Alt} />
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: Those strings "Payment Rejected" and "You canceled", can we put them on variables?

File: src/routes/chat/+page.svelte [hunk 12]
```diff
@@ -744,7 +804,47 @@
...
+                                                                    <Text hook="text-chat-message" markdown={"You canceled"} appearance={group.details.remote ? Appearance.Default : Appearance.Alt} />
+                                                                {/if}
+                                                            {/if}
+                                                        {:else if getValidPaymentRequest(line) !== undefined}
+                                                            {#if !$rejectedPayments.find(payments => payments.messageId === message.id)}
+                                                                {#if $own_user.key !== message.details.origin}
+                                                                    <Button hook="text-chat-message" class="send_coin" text={$_("payments.send_coin")} on:click={async () => getValidPaymentRequest(line, message.id)?.execute()}></Button>
+                                                                    <Button hook="text-chat-message" text={$_("payments.decline_payment")} appearance={Appearance.Error} on:click={async () => sendPaymentMessage(message, "reject")}>
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: reject here is magic string, let put in a variable as well.

File: src/routes/chat/+page.svelte [hunk 13]
```diff
@@ -744,7 +804,47 @@
...
+                                                                        <Icon icon={Shape.XMark}></Icon>
+                                                                    </Button>
+                                                                {:else if !checkForActiveRequest(message, line)}
+                                                                    <Button hook="text-chat-message" class="send_coin" text={$_("payments.sent_request")}></Button>
+                                                                    <Button hook="text-chat-message" text={$_("payments.cancel_request")} appearance={Appearance.Error} on:click={async () => sendPaymentMessage(message, "reject")}>
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: Same about "reject" string here.

File: src/routes/chat/+page.svelte [hunk 14]
```diff
@@ -744,7 +804,47 @@
...
+                                                                        <Icon icon={Shape.XMark}></Icon>
+                                                                    </Button>
+                                                                {:else}
+                                                                    <Button hook="text-chat-message" class="send_coin" text={$_("payments.sent_request")}></Button>
+                                                                    <Button hook="text-chat-message" text={$_("payments.canceled_request")} appearance={Appearance.Error} on:click={async () => sendPaymentMessage(message, "reject")}>
```
Comment (lgmarchi) [inline src/routes/chat/+page.svelte]: Same about "reject" string here.

**Review by lgmarchi — CHANGES_REQUESTED** [review]
I requested some changes to let code cleaner

**Review by phillsatellite — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
we shall address the rest of the feedback in separate PR's

**Review by lgmarchi — APPROVED** [review]

**Pull request merged**